pkgs/by-name/pa/pam_ssh_agent_auth/package.nix

```diff
@@ -54,6 +54,11 @@ stdenv.mkDerivation rec {
     "--with-cflags=-I$PWD"
   ];
 
+  env.NIX_CFLAGS_COMPILE = toString [
+    "-Wno-error=implicit-function-declaration"
+    "-Wno-error=incompatible-pointer-types"
+  ];
```

Comment:
Hmm. At the very least, this could use a comment explaining it, and a link to the upstream issue.

Debian has also created a patch that fixes the specific code that causes the error, which is probably better than disabling the errors globally.

https://sources.debian.org/src/pam-ssh-agent-auth/0.10.3-11/debian/patches/1000-gcc-14.patch/

I won't personally try to block anyone on switching approaches but in either approach it would be good to link to the [upstream issue](https://github.com/jbeverly/pam_ssh_agent_auth/issues/47) so that it will be understood in the future.

Reply:
Thanks for the pointer. Do you know if these patch urls are stable enough to fetch them, or is it customary to copy them over?

Reply:
It looks like it is stable enough, there are already a handful of packages with Debian patches being pulled in:

https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+sources.debian.org&type=code

In August of last year, someone added a helper for fetching Debian patches, [`fetchDebianPatch`](https://github.com/NixOS/nixpkgs/blob/e752b73238588a9959a3d89697e408cf90dca606/pkgs/build-support/fetchdebianpatch/default.nix)<sup>[[docs]](https://ryantm.github.io/nixpkgs/builders/fetchers/#fetchdebianpatch)</sup>:

https://github.com/NixOS/nixpkgs/blob/e752b73238588a9959a3d89697e408cf90dca606/pkgs/build-support/fetchdebianpatch/default.nix#L1-L24

It is less-utilized than just fetching directly from sources.debian.org, but is probably the cleaner way to go.

Reply:
Alright I've updated it to use `fetchDebianPatch` and got rid of the CFLAGS